### PR TITLE
[Framework] Add ACL counter related methods to `SonicHost`

### PR DESCRIPTION
**Pull request by lizhijianrd**
### Description of PR

Summary:
Added two ACL counter related methods to `SonicHost`. Test cases can use these methods to clear and read ACL counter.

**`get_acl_counter`**

```python
# sample code
def test_fun(duthosts, rand_one_dut_hostname):
    duthost = duthosts[rand_one_dut_hostname]
    packets_count = duthost.get_acl_counter('YOUR_ACL_TABLE', 'YOUR_ACL_RULE')
```

**`clear_acl_counters`**

```python
def test_fun(duthosts, rand_one_dut_hostname):
    duthost = duthosts[rand_one_dut_hostname]
    duthost.clear_acl_counters()
```

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### What is the motivation for this PR?

Added two ACL counter related methods to `SonicHost`. Test cases can use these methods to clear and read ACL counter.

#### How did you do it?

Add `get_acl_counter` and `clear_acl_counters` method to `SonicHost`.

#### How did you verify/test it?

Verified in some ACL testcases.

### CODE REVIEW

**Review by Blueve — APPROVED** [review]

**Comment by Blueve** [comment]
+ @bingwang-ms  to take a look on this new API. We hope we can reuse this method in test case and made our code more readable.

File: tests/common/devices/sonic.py [hunk 1]
```diff
@@ -2068,6 +2069,40 @@ def show_syslog(self):
         """
         return self.show_and_parse('show syslog')
 
+    def get_acl_counter(self, acl_table_name, acl_rule_name,
+                        timeout=ACL_COUNTERS_UPDATE_INTERVAL_IN_SEC * 3,
```
Comment (bingwang-ms) [inline tests/common/devices/sonic.py:2073]: ACL_COUNTERS_UPDATE_INTERVAL_IN_SEC * 2 should be enough?

File: tests/common/devices/sonic.py [hunk 2]
```diff
@@ -2068,6 +2069,40 @@ def show_syslog(self):
...
+            acl_table_name (str): Name of ACL table.
+            acl_rule_name (str): Name of ACL rule.
+            timeout (int): Maximum time (in second) wait for ACL counter available.
+            interval (int): Retry interval (in second) between read ACL counter.
+
+        Return:
+            packets_count (int): count of packets hit the specific ACL rule.
+        """
+        assert timeout >= 0 and interval > 0  # Validate arguments to avoid infinite loop
+        while timeout >= 0:
```
Comment (bingwang-ms) [inline tests/common/devices/sonic.py:2088]: How about add `aclshow -c` to clear the counter before reading the counter?

Comment (lizhijianrd) [reply tests/common/devices/sonic.py:2088]: Thanks @bingwang-ms! For this `get_acl_counter` function, I think we should return the ACL counter as it is, clear the counter may not be an expected behavior in all test scenario. We can add a separate `clear_acl_counter` function to apply the clear action. I'll update it later in ths PR.

File: tests/common/devices/sonic.py (hunk 1 above)
Comment (lizhijianrd) [reply tests/common/devices/sonic.py]: Thanks @bingwang-ms, I ran some ACL test on different platforms and I think `*2` is enough. Updated it in my latest commit.

**Pull request merged**